src/main/java/org/folio/rest/impl/AuthoritySourceFileAPI.java

```diff
@@ -65,8 +71,22 @@ public void putAuthoritySourceFilesById(String id, String lang,
                                           Map<String, String> okapiHeaders,
                                           Handler<AsyncResult<Response>> asyncResultHandler,
                                           Context vertxContext) {
+    normalizeBaseUrl(entity);
+
     PgUtil.put(REFERENCE_TABLE, entity, id, okapiHeaders, vertxContext,
       PutAuthoritySourceFilesByIdResponse.class,
       asyncResultHandler);
   }
```

Comment:
What about this requirement? 
"Predefined Base URL can be overridden by a user with permission to do so AND that override can be set to null "

Maybe PATCH endpoint?

Reply:
And also it should be restricted to edit other fields or delete predefined entities. I think we can use your `source` field to identify what can be edited or deleted.

Reply:
> "Predefined Base URL can be overridden by a user with permission to do so AND that override can be set to null "
Maybe PATCH endpoint?

> And also it should be restricted to edit other fields or delete predefined entities. I think we can use your `source` field to identify what can be edited or deleted.


@psmagin Anyway we should keep PUT with possibility to modify the whole entity as it is used by [TenantLoading](https://github.com/folio-org/raml-module-builder/blob/master/domain-models-runtime/src/main/java/org/folio/rest/tools/utils/TenantLoading.java) service when reference data needs to be updated. I've discussed this with Khalila, seems that she is ok with restricting modification on UI side (as it's done for some other resources i.e. [classificationTypes](https://folio-snapshot.dev.folio.org/settings/inventory/classificationTypes)). The discussion is here [MODINVSTOR-892](https://issues.folio.org/browse/MODINVSTOR-892)